event-sourcing/subscription-manager/src/main/java/uk/gov/justice/services/event/sourcing/subscription/jmx/MBeanInstantiator.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.services.event.sourcing.subscription.jmx;
+
+
+import static java.lang.management.ManagementFactory.getPlatformMBeanServer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+@Singleton
+@Startup
+public class MBeanInstantiator {
+
+    private static final String SHUTTERING_DOMAIN_NAME = "shuttering";
+    private static final String CATCHUP_DOMAIN_NAME = "catchup";
+
+    private static final String SHUTTERING_BEAN = "DefaultShutteringMBean";
+    private static final String CATCHUP_BEAN = "DefaultCatchupMBean";
+
+    private Map<Object, ObjectName> mbeanMap = createMBeanMap();
```

Comment:
Maybe we can make this more extensible by injecting an object that contains the map of MBeans to register